src/librustc_driver/pretty.rs

```diff
@@ -347,7 +346,7 @@ impl<'a, 'tcx> pprust_hir::PpAnn for TypedAnnotation<'a, 'tcx> {
             s.s.space();
             s.s.word("as");
             s.s.space();
-            s.s.word(self.tables.get().expr_ty(expr).to_string());
+            s.s.word(self.tables.get().unwrap().expr_ty(expr).to_string());
```

Comment:
For `hir::Expr`/`hir::Pat` this is easy: they cannot exist outside a `hir::Body`. It's only hard to reason about `hir::Ty` (or `hir::QPath`/`hir::Path` because those can be from `hir::Ty` or I guess a `hir::TraitRef`).